### PR TITLE
feat(composite-client): try to get pending txs from peers first

### PR DESCRIPTION
**Pull request by karlprieb**

### CODE REVIEW

File: src/arweave/composite-client.ts [hunk 1]
```diff
@@ -415,24 +415,14 @@ export class ArweaveCompositeClient
...
-              url: `/unconfirmed_tx/${txId}`,
-            })
-            .then((response) => {
-              return response.data;
-            });
-        }
-
         return this.peerGetTx(txId)
           .catch(async () => {
+            const url = `/${isPendingTx ? 'unconfirmed_tx' : 'tx'}/${txId}`;
```
Comment (djwhitt) [inline src/arweave/composite-client.ts:420]: We need this same logic `peerGetTx`: https://github.com/ar-io/ar-io-node/blob/ec9f2f038d4c67f74e2289d1a46210142179252d/src/arweave/composite-client.ts#L379

**Review by djwhitt — CHANGES_REQUESTED** [review]
This simplifies the request from the trusted node, but it still needs an adjustment to fetch from peers.

File: src/arweave/composite-client.ts [hunk 2]
```diff
@@ -415,24 +415,14 @@ export class ArweaveCompositeClient
...
+
             // Request TX from trusted node if peer fetch failed
             return this.trustedNodeRequestQueue.push({
               method: 'GET',
-              url: `/tx/${txId}`,
+              url,
```
[the remark this answers is not in the document]
Comment (djwhitt) [reply src/arweave/composite-client.ts:425]: Not bad. This is actually a reasonable suggestion.

File: src/arweave/composite-client.ts (hunk 1 above)
Comment (karlprieb) [reply src/arweave/composite-client.ts]: Fixed here [`f01b214` (#119)](https://github.com/ar-io/ar-io-node/pull/119/commits/f01b214c2d361e2de70b806def6216b55adf88e5)

**Review by djwhitt — APPROVED** [review]

**Pull request merged**